### DISCUSSION TITLE
REST API with api/server/configuration

### DISCUSSION THREAD

**Discussion by durantschoon** [nextgenhealthcare/connect#4677] [Q&A] [answered]
version is 3.11.0

Hi,

We're experimenting with setting up automated testing and want to configure a running instance of NextGen Connect with a predefined xml file. We are able to successfully POST the file and many aspects appear to work:

1. Expected channels appear
2. Channels are deployed as expected when using "?deploy=true"

But there are some problems we've been experiencing. For example, the Javascript transformer code comes in without newlines. So scripts that use inline comments (`// like this`) will comment out the rest of the script. We think we can use multiline comments on a single line (`/* like this */`) and the code will work, but in general the channel transformers are not working, even if they are of type "Message Builder" and not "JavaScript". This in contrast to using "Settings > Restore Config" and loading the same file. In the manual loading case, the channels work as expected and the newlines in the scripts are retained. 

I did a bit of debugging on my own, but haven't come up with anything. I'm wondering if anyone in the community has helpful experience or advice.

We're launching this kind of command in Unix where $XML_SERVER_CONFIG evaluates to the xml configuration file

```sh
curl \
    -k \
    -X PUT \
    -H "Content-Type: application/xml" \
    -H "Authorization: Basic $AUTH_TOKEN" \
    --data @$XML_SERVER_CONFIG \
    "https://127.0.0.1:8443/api/server/configuration?deploy=true"
```

Thanks in advance,

**Comment by tonygermano** [comment, marked as answer]
It's likely curl that is causing the problem.

See https://github.com/nextgenhealthcare/connect/issues/4391

**Reply by durantschoon** [reply]
Oh thank you! ... early tests show this is the likely culprit with `--data-binary` fixing it.
